### DISCUSSION TITLE
Structure Should be Optional

### DISCUSSION THREAD

**Discussion by Deviltree** [BSData/adeptus-titanicus#93] [Ideas]
I have received a ton of feedback and the most common request I've gotten across in person, in discord, and across forums is to be able to hide structure.  I think if we just change it from being mandatory, people who still want to use it can.

This is also related to us being able to make a streamlined out put so we can easily share Lists.

**Comment by ghost** [comment]
Deleting the Show Structure item does this.

It has to be one by default at list creation, though, otherwise it takes dozens and dozens of clicks to add back, so there is less burden when initially set on.

I went looking (on Facebook groups, Reddit, the relatively new BSData Discord) and couldn't find the feedback you're describing. Can you link me to some?

**Comment by TrentBartlem** [comment]
I personally use both; sometimes I want it, other times it's irrelevant to me. 
Two things that do annoy me are

* it's an uncategorised switch (edit: see https://imgur.com/a/H37ER0k)
* it doesn't turn into a radio button when clicked, like the options that kill team has

**Comment by amis92** [comment]
Does it have any use except being an in-game tracker? As in, does it have use in pre-game list building?

I'd like to emphasize that BS data files are primarily meant as roster building. Wound/other-in-game-item tracking items are very much against nature of how the app works.

I've tried out how it looks, I think I've used updated data. The way it is, Show Structure doesn't work as expected. Removing it doesn't remove all Structure entries. It marks them as hidden, which only puts an additional error per Titan. So a user actually has to go and delete every Structure of every Titan to get rid of in in the output.

As such, I'd advise to either:
- make Structure an opt-in system reference system
- make `Show Structure` switch actually hide all those entries without needing to remove them (possibly not possible, since you have to select the entry with the profile in it for the tracker to work as intended).

PS Every data dev wishes we had tracking options for our datafiles, but it's a pain in the ass for everyone who doesn't wish to use it in a given instance - and actually, that's most of the app usage. People love to play with lists, share them, discuss etc. Only a small percentage of lists are actually built and used for the game. I can't share data, but I did talk with Jonski about that last year.

**Comment by ghost** [comment]
While removing it does throw one error per titan, adding it in (if the default state was hidden) is many more clicks, because of the opinionated way that the defaults are not set on hidden items.

**Comment by amis92** [comment]
Do take into account the ratio of rosters for fun/checking options/verifying limits/sharing with others to rosters used in games, and it turns out you'd save much more clicks/taps if it were opt-in.

We work in a constrained environment. We can't always validate everything, and even if we could it's sometimes not worth it. We also can't change the app, so we have to adjust datafiles to how it works out for people. The last part may change with Phalanx but it's a song of the future.

**Reply by ghost** [reply]
Thinking about this, I wonder if a good solution would be to include an XSLT or CSS sheet to hide these things for printout with instructions for users in how to do so, if indeed the print use case is really the source of the complaint?

**Reply by amis92** [reply]
@capitaladot XSLT/CSS is not an option for mobile. Mobile is over 90% of activity(subs), hard fact from Jonski. Structure is hardest on mobile.

On PC you can select or deselect things 10x quicker due to having a tree. What you're suggesting is just increasing comfort on an already quite comfortable device. The issue is mainly on mobile.

*It's always the mobile. Nested choices are bad, too much back-forth is back, this bad, that bad. It's the worst platform and the most used one.* 🤮

I think you can go on forums, facebook etc and ask people for opinion. Obviously in an unopinionated way, 'cause people most love to troll and hate and not give thumbs up. :D
